src/tools/rbd/action/DiskUsage.cc

```diff
@@ -155,6 +155,11 @@ static int do_disk_usage(librbd::RBD &rbd, librados::IoCtx &io_ctx,
       continue;
     }
 
+    snap_list.erase(remove_if(snap_list.begin(),
```

Comment:
@dillaman  Shall we display snapshots from all namesapce? If so, we can add a `--all` flag here, and another bug needs to be fixed (cause of `librbd::image::OpenRequest::send_set_snap()` and `librbd::api::DiffIterate<>::diff_iterate()` always passing UserSnapshotNamespace):
```
root@s222:/ceph-dev/build# rbd snap ls img1 -a
SNAPID NAME                                      SIZE TIMESTAMP                NAMESPACE            
     4 6ae85382-edf6-4bb9-8e70-6b855c4dde6a 102400 kB Thu Feb 22 16:39:45 2018 trash                
     5 .group.3_10542ae8944a_104d3d1b58ba   102400 kB Thu Feb 22 16:46:21 2018 group (rbd/g1@sanp1)
root@s222:/ceph-dev/build# rbd du img1        
2018-02-22 16:58:10.891 7f06d17fa700 -1 librbd::image::OpenRequest: failed to set image snapshot: (2) No such file or directory
rbd: error opening snapshot img1@6ae85382-edf6-4bb9-8e70-6b855c4dde6a: (2) No such file or directory
NAME PROVISIONED USED 
```

Reply:
Such a change would require new API methods to set a snapshot by ID (instead of name) and ```diff_iterate``` by snapshot id. If you are interested, please feel free to add since I think it would be valuable.